File: Ruby/Guias/G7/Ej5/checking_account.rb
require_relative 'Accounts'
class CheckingAccount < Accounts
  def initialize(id,overdraft)
    @id = id
    @overdraft = overdraft
    @balance=0
  end
  def withdraw(amount)
    raise 'No cuenta con los fondos necesarios' unless self.withdraw?(amount)
    super(amount)
  end

  protected def withdraw?(amount)
    @balance + @overdraft - amount >= 0
  end
end

File: Java/src/Guias/TP3/Ej7/CheckingAccount.java
package Guias.TP3.Ej7;

public class CheckingAccount extends Accounts{
    private final double overDraft;
    CheckingAccount(int id, double overDraft){
        super(id);
        this.overDraft=overDraft;
    }

    @Override
    protected boolean canExtract(double amount) {
        return getBalance()+this.overDraft-amount>=0;
    }

    public double getOverDraft() {
        return overDraft;
    }
}
